service/src/main/java/org/jboss/sbomer/service/nextgen/service/rest/v1beta2/EventsApi.java

```diff
@@ -99,14 +107,35 @@ public class EventsApi {
             responseCode = "500",
             description = "Internal server error",
             content = @Content(mediaType = MediaType.APPLICATION_JSON))
-    public Response search(@Valid @BeanParam PaginationParameters paginationParams) {
-        List<EventRecord> events = Event.findAll()
+    public Response search(@Valid @BeanParam PaginationParameters paginationParams, @QueryParam("query") String query) {
+
+        PanacheQuery<Event> panacheQuery;
+
+        if (query == null || query.isBlank()) {
+            panacheQuery = Event.findAll();
+        } else {
+            try {
+                EventsQueryListener listener = queryProcessor.process(query);
```

Comment:
It is OK to have this here for now, for testing purposes, but we need to think how to make it cleaner later.